    @Test
    public void testDelete() throws IOException
    {
        DeletingArchiver da = new DeletingArchiver(45);
        List<File> files = generateFiles(10, 5);
        for (File f : files)
            da.onReleased(1, f);
        // adding 5 files, each with size 10, this means the first one should have been deleted:
        assertFalse(files.get(0).exists());
        for (int i = 1; i < files.size(); i++)
            assertTrue(files.get(i).exists());
        assertEquals(40, da.getBytesInStoreFiles());
    }

